mobile/lib/src/application/settings/settings_cubit.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:polydodo/src/common/constants.dart';
+import 'package:polydodo/src/domain/settings/i_settings_repository.dart';
+import 'package:polydodo/src/domain/settings/settings.dart';
+
+part 'settings_state.dart';
+
+class SettingsCubit extends Cubit<SettingsState> {
+  final ISettingsRepository _repository;
+
+  SettingsCubit(this._repository) : super(SettingsLoadInProgress()) {
+    getSettings();
+  }
+
+  Future<void> getSettings() async {
+    var settings = {
+      AGE_KEY: await _repository.getSetting(AGE_KEY),
+      SERVER_URL_KEY: await _repository.getSetting(SERVER_URL_KEY) ?? 'Not Set',
+      SEX_KEY:
+          Sex.values[await _repository.getSetting(SEX_KEY)] ?? Sex.NotSet.index,
+    };
```

Comment:
@WilliamHarvey97 @cicero-mt  I would even argue the Cubit shouldn't know about these settings. They should be fetch as the UI is created, thus removing code duplication (for instance, AGE_KEY is mentionned in 2 different places where it should only be referenced once).

Reply:
Je sais pas, sinon c'est le ui qui set le type des options stockées au lieu d'être la logique.

Reply:
Alternative: the cubit knows about them, but we need to add more information about how they should be handled and the UI just loops these settings (without referencing one specific setting of course).

Reply:
settings should be an entity in our domain layer

Reply:
The cubit should not know what are the key. This is something specific to our persistency schema. Thus, it is the role of the DAL to keep these keys. It should be kept secret for the app and presentation layer (they only know about the domain and its Settings entity that can be retrieve from Future<Settings>ISettingsRepository::read()).